pyaerocom/io/readgridded.py

```diff
@@ -364,15 +363,15 @@ def _get_vars_provided(self):
         _vars = []
         _vars.extend(self.vars_filename)
 
-        for aux_var in self.AUX_REQUIRES.keys():
+        for aux_var in self.AUX_REQUIRES:
             if "*" in aux_var:
                 continue
-            elif not aux_var in _vars and self.check_compute_var(aux_var):
+            if not aux_var in _vars and self.check_compute_var(aux_var):
```

Comment:
Is there a special reasom you got rid of the `elif` here?

Reply:
IMO, an `if` clause ending in a `continue`, `break`, `return` or `raise` is a guarding clause.
The same applies to the `elif` clause here.

Think about the following function:

``` python
def float_division(numerator: float, denominator: float) -> float:
    if not isinstance(numerator, float):
        raise TypeError("numerator should be a float")
    elif not isinstance(denominator, float):
        raise TypeError("denominator should be a float")       
    elif denominator == 0:
        raise ValueError("dude you can not divide by zero")
    else:
        return numerator/denominator
```

I find it more readable with the `if/elif` as a guarding clauses, indicating the anomalous cases,
and the "expected use" in the body of the function instead of an `else` clause

``` python
def float_division(numerator: float, denominator: float) -> float:
    if not isinstance(numerator, float):
        raise TypeError("numerator should be a float")
    if not isinstance(denominator, float):
        raise TypeError("denominator should be a float")       
    if denominator == 0:
        raise ValueError("dude you can not divide by zero")

    return numerator/denominator
```